src/components/dashboard/languageToggle.tsx

```diff
@@ -1,56 +1,38 @@
 'use client'
-import React, { useEffect, useState } from 'react'
-import { useTranslation } from 'react-i18next'
+import React, { useState } from 'react'
 import {
   Select,
   SelectContent,
   SelectItem,
   SelectTrigger,
   SelectValue,
 } from '@/components/ui/select'
-import { changeLanguage } from '@/i18n'
-
-type Language = {
-  nativeName: string
-}
-
-type Languages = {
-  [key: string]: Language
-}
-
-const lngs: Languages = {
-  en: { nativeName: 'English' },
-  tr: { nativeName: 'Türkçe' },
-}
+import { supportedLanguages } from '@/lib/languageDetector'
+import i18next from 'i18next'
 
 function LanguageToggle(): JSX.Element {
-  const { i18n } = useTranslation()
-  const [selectedLanguage, setSelectedLanguage] = useState(i18n.language)
-
-  useEffect(() => {
-    const savedLanguage = localStorage.getItem('language') || 'tr'
-    if (i18n.language !== savedLanguage) {
-      i18n.changeLanguage(savedLanguage)
+  const [lang, setLang] = useState(i18next.language)
+  const changeLanguage = (lng: string): void => {
+    i18next.changeLanguage(lng).then((r) => r)
+    if (typeof window !== 'undefined') {
```

Comment:
i18next.changeLanguage(lng).then((r) => r)

bu satır ne işe yarıyor? "r" ne?

Reply:
Bu kısmı kullanmasam da olur. Promise'in çözülmesini beklemek için ekledim. i18n/index.ts dosyasında eklediğim için buraya da ekledim. 
Bunun yerine 
```
i18next
      .changeLanguage(language)
      .catch((error) => console.log(error.message()))
``` bu şekilde de yazabilirim. Hangisi olsun?

Reply:
kullanmasak da oluyosa kullanmayalım